### PR TITLE
Feat/schedule autoscale

### PR DESCRIPTION
**Pull request by crgisch**

### CODE REVIEW

File: provision/kubernetes/autoscale.go [hunk 1]
```diff
@@ -22,6 +23,7 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 	vpav1 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
 	vpaclientset "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/client/clientset/versioned"
+	k8sutilsptr "k8s.io/utils/ptr"
```
Comment (wpjunior) [inline provision/kubernetes/autoscale.go:26]: using `ptr.To` is a good choice.

```suggest
"k8s.io/utils/ptr"
```

File: provision/kubernetes/autoscale.go [hunk 2]
```diff
@@ -144,18 +152,61 @@ func (p *kubernetesProvisioner) GetAutoScale(ctx context.Context, a provision.Ap
...
+			}
+			specs = append(specs, scaledObjectToSpec(*observedKEDAScaledObject))
+		} else {
+			specs = append(specs, hpaToSpec(*hpa))
+		}
 	}
 	return specs, nil
 }
 
+func KEDAScaledObjectName(hpa autoscalingv2.HorizontalPodAutoscaler) string {
```
Comment (wpjunior) [inline provision/kubernetes/autoscale.go:178]: @crgisch this func should be private, rename to kedaScaledObjectName

File: provision/kubernetes/autoscale.go [hunk 3]
```diff
@@ -219,15 +270,39 @@ func (p *kubernetesProvisioner) RemoveAutoScale(ctx context.Context, a provision
...
+
+	err = client.AutoscalingV2().HorizontalPodAutoscalers(ns).Delete(ctx, hpaName, metav1.DeleteOptions{})
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func RemoveKEDAScaleObject(ctx context.Context, client *ClusterClient, ns string, scaledObjectName string) error {
```
Comment (wpjunior) [inline provision/kubernetes/autoscale.go:299]: @crgisch you could rename RemoveKEDAScaleObject to removeKEDAScaleObject

File: provision/provision.go [hunk 4]
```diff
@@ -511,9 +519,6 @@ func (s AutoScaleSpec) ToCPUValue(a App) (int, error) {
 }
 
 func (s AutoScaleSpec) Validate(quotaLimit int, a App) error {
-	if s.MinUnits == 0 {
```
Comment (wpjunior) [inline provision/provision.go:514]: @crgisch why you removed this validation?

Comment (crgisch) [reply provision/provision.go:514]: the ideia here was to allow setting min: 0 in order to allow schedule to downscale apps to 0

**Review by wpjunior — APPROVED** [review]

**Pull request merged**